tools/run-dev.py

```diff
@@ -341,7 +341,9 @@ def shutdown_handler(*args: Any, **kwargs: Any) -> None:
 
 def print_listeners() -> None:
     external_host = os.getenv('EXTERNAL_HOST', 'localhost')
-    print(f"\nStarting Zulip on:\n\n\t{CYAN}http://{external_host}:{proxy_port}/{ENDC}\n\nInternal ports:")
+    if not str(proxy_port) in external_host:
+        external_host += f':{proxy_port}'
+    print(f"\nStarting Zulip on:\n\n\t{CYAN}http://{external_host}/{ENDC}\n\nInternal ports:")
```

Comment:
This substring check could incorrectly match anywhere in the string, and this logic isn’t quite right anyway. `EXTERNAL_HOST` should always include the port if it’s needed, so I think we simply want

```python
    external_host = os.getenv('EXTERNAL_HOST', f'localhost:{proxy_port}')
    print(f"\nStarting Zulip on:\n\n\t{CYAN}http://{external_host}/{ENDC}\n\nInternal ports:")
```